src/docks/encodedock.cpp

```diff
@@ -700,6 +700,77 @@ Mlt::Properties *EncodeDock::collectProperties(int realtime, bool includeProfile
                 setIfNotSet(p, "g", ui->gopSpinner->value());
                 setIfNotSet(p, "bf", ui->bFramesSpinner->value());
                 p->set("x265-params", x265params.toUtf8().constData());
+            } else if (vcodec == "libsvtav1") {
+                // Most SVT-AV1 parameters must be supplied through svtav1-params.
+                // The preset properties are for UI controls and custom presets.
+                QString bitrate_text = ui->videoBitrateCombo->currentText();
+                int bitrate_kbps = qMax(QString(bitrate_text).replace('k', "").replace('M', "000").toInt(), 1);
+                int buffer_bits = qRound(ui->videoBufferSizeSpinner->value() * 1024.0 * 8.0);
+                int buffer_ms = qRound(float(buffer_bits) / float(bitrate_kbps));
+                QStringList encParams;
+
+                switch (ui->videoRateControlCombo->currentIndex()) {
+                case RateControlAverage: {
+                    encParams << QString("rc=1");
+                    encParams << QString("tbr=%1").arg(bitrate_text);
+                    setIfNotSet(p, "vb", bitrate_text.toLatin1().constData());
+                    break;
+                }
+                case RateControlConstant: {
+                    encParams << QString("rc=2");
+                    encParams << QString("tbr=%1").arg(bitrate_text);
+                    encParams << QString("buf-sz=%1").arg(buffer_ms);
+                    setIfNotSet(p, "vb", bitrate_text.toLatin1().constData());
+                    setIfNotSet(p, "vbufsize", buffer_bits);
```

Comment:
**Summary:**
I think the FFmpeg wrapper is buggy for signaling CBR. I think the `encodedock.cpp` code for CBR mode may need to remain as-is. There are Reddit posts from 11 days ago regarding this problem.

**Detail:**
Here is the code from `libavcodec/libsvtav1.c`:

```c
    if (avctx->bit_rate) {
        param->target_bit_rate      = avctx->bit_rate;
        if (avctx->rc_max_rate != avctx->bit_rate)
            param->rate_control_mode = 1;
        else
            param->rate_control_mode = 2;
```

If `vmaxrate` is not specified, then VBR mode is chosen. If `vmaxrate` is specified (the value does not matter), then SVT-AV1 throws this error:

```text
Svt[error]: Instance 1: Max Bitrate only supported with CRF mode
```

Since `vmaxrate` cannot be specified without an error, there is no way to trigger the wrapper to set `rate_control_mode = 2` (CBR). The workaround that other people are doing is to set `rc=2` in `svtav1-params`, hence my suggestion to leave this code as-is.

Reply:
I agree and good find!